sql/cagg_migrate.sql

```diff
@@ -544,6 +562,12 @@ BEGIN
     CALL _timescaledb_internal.cagg_migrate_create_plan(_cagg_data, _cagg_name_new, override, drop_old);
     COMMIT;
 
+    -- SET LOCAL is only active until end of transaction.
+    -- While we could use SET at the start of the function we do not
+    -- want to bleed out search_path to caller, so we do SET LOCAL
+    -- again after COMMIT
+    SET LOCAL search_path TO pg_catalog, pg_temp;
+
```

Comment:
Same here, does this not mean that the search path is set in the transaction starting here? There is no implicit commit at the end of the procedure, so this potentially leaks out to the surrounding transaction.

Reply:
Are u meant if, for example, the user explicit start a transaction and execute the cagg_migrate?

Reply:
Yes, for example, but it would apply also if you do not start a transaction.

```
mats=# create procedure foo() as $$ begin set search_path to magic; end $$ language plpgsql;
CREATE PROCEDURE
mats=# show search_path;
   search_path   
-----------------
 "$user", public
(1 row)

mats=# call foo();
CALL
mats=# show search_path;
 search_path 
-------------
 magic
(1 row)
```

Reply:
@mkindahl your example is not entirely correct because you are not using `SET LOCAL`. See the example above:

Reply:
Need to reset the search_path at the end of the procedure execution, for example:
```sql
fabrizio=# create procedure foo() as $$ begin set local search_path to magic; raise info 'foo'; reset search_path; end $$ language plpgsql;
CREATE PROCEDURE

fabrizio=# show search_path ;
   search_path   
-----------------
 "$user", public
(1 row)

fabrizio=# begin ;
BEGIN
fabrizio=*# call foo();
INFO:  foo
CALL
fabrizio=*# show search_path ;
   search_path   
-----------------
 "$user", public
(1 row)

fabrizio=*# rollback ;
ROLLBACK
fabrizio=# show search_path ;
   search_path   
-----------------
 "$user", public
(1 row)
```

Reply:
@mkindahl actually users can't execute the `cagg_migrate` in an explicit transaction because we have transaction control statements inside:
```sql
BEGIN;
CALL cagg_migrate('conditions_summary_daily');
psql:include/cagg_migrate_common.sql:349: ERROR:  invalid transaction termination
```
From the Postgres documentation: https://www.postgresql.org/docs/current/sql-call.html
_If CALL is executed in a transaction block, then the called procedure cannot execute transaction control statements. Transaction control statements are only allowed if CALL is executed in its own transaction._

Reply:
> @mkindahl actually users can't execute the cagg_migrate in an explicit transaction because we have transaction control statements inside:

But the `search_path` can still leak into the surrounding environment as illustrated above.

Reply:
@mkindahl to resume it will never happen... so added some additional tests to demonstrate it.

Reply:
> > @mkindahl actually users can't execute the cagg_migrate in an explicit transaction because we have transaction control statements inside:
> 
> But the `search_path` can still leak into the surrounding environment as illustrated above.

Not in the case where we use transaction statements inside the procedure:
```sql
fabrizio=# create procedure foo() as $$ begin set local search_path to magic; raise info 'foo'; commit; reset search_path; end $$ language plpgsql;
CREATE PROCEDURE
fabrizio=# show search_path ;
   search_path   
-----------------
 "$user", public
(1 row)

fabrizio=# begin;
BEGIN
fabrizio=*# call foo();
INFO:  foo
ERROR:  invalid transaction termination
CONTEXT:  PL/pgSQL function foo() line 1 at COMMIT
```

Reply:
> Not in the case where we use transaction statements inside the procedure:

No, but in the case where you do *not* use an explicit transaction. Right?

Reply:
> > Not in the case where we use transaction statements inside the procedure:
> 
> No, but in the case where you do _not_ use an explicit transaction. Right?

In the case we don't use explicit transaction the `SET LOCAL search_path` will be visible just inside the procedure. It will not leak as I demonstrated here: https://github.com/timescale/timescaledb/pull/4935#discussion_r1016635278

Reply:
> > > Not in the case where we use transaction statements inside the procedure:
> > 
> > 
> > No, but in the case where you do _not_ use an explicit transaction. Right?
> 
> In the case we don't use explicit transaction the `SET LOCAL search_path` will be visible just inside the procedure. It will not leak as I demonstrated here: [#4935 (comment)](https://github.com/timescale/timescaledb/pull/4935#discussion_r1016635278)

Ah, missed that comment. Then all is good.